main.c

```diff
@@ -37,70 +37,82 @@ int main()
         }
-        Bcachefs_extent extent = Bcachefs_iter_make_extent(&bchfs, &bchfs_iter);
-        printf("extent: i:%llu fo:%llu, o:%llu, s:%llu\n",
-               extent.inode, extent.file_offset, extent.offset, extent.size);
-        switch (bkey->type)
+        Bcachefs_iter_fini(&bchfs, &bchfs_iter);
+    }
+    
+    // Dirent
+    // -----------------------------------------------------------------------------
+    {
+        Bcachefs_iterator bchfs_iter = {0};
+        Bcachefs_iter(&bchfs, &bchfs_iter, BTREE_ID_dirents);
+        bch_val = Bcachefs_iter_next(&bchfs, &bchfs_iter);
+        int i = 0;
+        while (bch_val)
         {
-        case KEY_TYPE_extent:
-            printf("extent: i:%llu fo:%llu, o:%llu, s:%llu\n",
-                   extent.inode, extent.file_offset, extent.offset, extent.size);
+            Bcachefs_dirent dirent = Bcachefs_iter_make_dirent(&bchfs, &bchfs_iter);
+            char fname[30] = {0};
+            memcpy(fname, dirent.name, dirent.name_len);
 
-            fseek(bchfs.fp, (long)extent.offset, SEEK_SET);
-            uint8_t *bytes = malloc(extent.size);
-            fread(bytes, extent.size, 1, bchfs.fp);
-            printf("file: n:%s, t:%ld\n", fname, ftell(fp));
-            fseek(fp, (long)extent.file_offset, SEEK_SET);
-            printf("file: n:%s, t:%ld\n", fname, ftell(fp));
-            fwrite(bytes, extent.size, 1, fp);
-            break;
-        case KEY_TYPE_inline_data:
-            printf("extent: i:%llu fo:%llu, o:%llu, s:%llu\n",
-                   extent.inode, extent.file_offset, extent.offset, extent.size);
-            printf("d:[%s]\n", (const uint8_t*)bch_val);
+            printf("dirent %3d: p:%10llu, i:%10llu, t:%10u, %s\n",
+                i,
+                dirent.parent_inode,
+                dirent.inode,
+                dirent.type,
+                fname);
 
-            printf("file: n:%s, t:%ld\n", fname, ftell(fp));
-            fseek(fp, (long)extent.file_offset, SEEK_SET);
-            printf("file: n:%s, t:%ld\n", fname, ftell(fp));
-            fwrite(bch_val, extent.size, 1, fp);
-            break;
+            bch_val = Bcachefs_iter_next(&bchfs, &bchfs_iter);
+            i += 1;
         }
-        fclose(fp);
-    }
-    Bcachefs_iter_fini(&bchfs, &bchfs_iter);
-    Bcachefs_iter(&bchfs, &bchfs_iter, BTREE_ID_dirents);
-    bch_val = Bcachefs_iter_next(&bchfs, &bchfs_iter);
-    for (; bch_val; bch_val = Bcachefs_iter_next(&bchfs, &bchfs_iter))
-    {
-        Bcachefs_dirent dirent = Bcachefs_iter_make_dirent(&bchfs, &bchfs_iter);
-        char fname[30] = {0};
-        memcpy(fname, dirent.name, dirent.name_len);
-        printf("dirent: p:%llu, i:%llu, t:%u, %s\n",
-               dirent.parent_inode,
-               dirent.inode,
-               dirent.type,
-               fname);
+        Bcachefs_iter_fini(&bchfs, &bchfs_iter);
     }
-    Bcachefs_iter_fini(&bchfs, &bchfs_iter);
+
     Bcachefs_fini(&bchfs);
     return 0;
 }
+
```

Comment:
Extra line at end of file (github doesn't show the last line)
```suggestion
}
```

Reply:
I agree with this standard but in this case there are 2 empty lines at the end:
<img width="378" alt="Capture d’écran, le 2021-10-18 à 16 33 04" src="https://user-images.githubusercontent.com/6757005/137802833-b588f421-8949-405d-a7c5-b9b27979fce9.png">

That's why I'm mentioning that there is an extra line

Reply:
```suggestion
}
```